File: src/server/components/menu.rs
use crate::views;

pub fn menu(path: &str) -> String {
    let items: Vec<String> = vec![
        views::navbar_item("/", path == "/", "Posts"),
        views::navbar_item("/albums", path == "/albums", "Albums"),
        views::navbar_item("/counter", path == "/counter", "Counter"),
    ];

    views::navbar(&items.join(""))
}

File: src/client/components/menu.ts
import { $navbar } from "@/client/views/navbar";
import { $navbar_item } from "@/client/views/navbar-item";

export function menu(path: string) {
  return $navbar([
    $navbar_item("/", path === "/", "Posts"),
    $navbar_item("/albums", path === "/albums", "Albums"),
    $navbar_item("/counter", path === "/counter", "Counter"),
  ]);
}